ldms/src/sampler/lustre_client/lustre_client.c

```diff
@@ -227,6 +227,15 @@ static int config(struct ldmsd_plugin *self,
                   struct attr_value_list *kwl, struct attr_value_list *avl)
 {
         log_fn(LDMSD_LDEBUG, SAMP" config() called\n");
+	char *ival = av_value(avl, "instance");
+	if (ival) {
+		if (strlen(ival) < sizeof(producer_name)) {
+			strncpy(producer_name, ival, sizeof(producer_name));
```

Comment:
Why are you putting the "instance" name in the "producer name" variable?  This would also change thre producer when the instance is set. I think that would be unexpected. I think you need to add an instance_name variable, add that to the parameters passed to llite_general_create(), and merge things there.

Reply:
The "custom" of ldms deployment is that for a given schema, the set instance name is scripted as $producer/$schema (or at least $producer/$plugin).  
Where there are multiple sets per plugin (as this case, or ibnet, or ibmad), prescribing the entire instance name in the configuration script is nonsensical, so we use the 'instance' as a prefix instead of the entire name: $instance/$device_name.
The convention for $producer is $hostname, but for scale testing where many sampler daemons run on the same node, the sets must be distinguishable so "producer_name[]" must be some extension/replacement on hostname (e.g. localhost[1-n] or $hostname/$k). 

This all fits well with the C handling of the "instance" parameter as a prefix instead of inventing a new configuration item (instance_prefix=) and pushing around extra arguments.
The same prefix applies to all the set instances, unless you want to call config twice.

Reply:
I don't understand what that explanation had to do with my question. I am fine with you having an "instance" configurable, but the "instance" configurable should not be changing the "producer" field. The "producer" configurable should change the producer field.

And something seems a little fishy here to me about the currently implementation breaking a test environment. The code is currently setting the instance name to "producer/llite_name", so the only way those would conflict in a testing environment is if the testing environment is failing to correctly configure different producer names for different producers on the same node.

Reply:
ah. so you want 's/instance/producer/g' in this change afa the config input. I'll do that.

Reply:
I just don't want the instance configurable to change the producer name. :)

But, yes, it sounds like making the producer name configurable is probably what you really needed rather than making the instance name directly configurable. One could potential make both configurable, but it sounds like producer might be sufficient for your needs.

Reply:
revised to producer option.